scripts/readelf.py

```diff
@@ -1332,6 +1335,58 @@ def _dump_debug_frames_interp(self):
                     self._dwarfinfo.debug_frame_sec,
                     self._dwarfinfo.CFI_entries())
 
+    def _dump_debug_locations(self):
+        """ Dump the location lists from .debug_location section
+        """
+        def _get_cu_base(cu):
+            top_die = cu.get_top_DIE()
+            attr = top_die.attributes
+            if 'DW_AT_low_pc' in attr:
+                return attr['DW_AT_low_pc'].value
+            elif 'DW_AT_entry_pc' in attr:
+                return attr['DW_AT_entry_pc'].value
+            else:
+                raise ValueError("Can't find the base IP (low_pc) for a CU")                    
+
+        di = self._dwarfinfo
+        loc_lists = di.location_lists()
+        if loc_lists:
+            loc_lists = [ll for ll in loc_lists.iter_location_lists()]
+            if len(loc_lists) > 0:
+                # To dump a location list, one needs to know the CU.
+                # Scroll through DIEs once, list the known location list offsets
+                CUs = [cu for cu in di.iter_CUs()]
```

Comment:
```CUs = list(di.iter_CUs())```? 

But do you actually use it for random access? Why not iterate over `iter_CUs` directly?

Reply:
Good point.